Update ExoPlayer to 2.18.0

TacoTheDank opened the pull request:
#### What is it?
- [ ] Bugfix (user facing)
- [ ] Feature (user facing)
- [x] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR
- Updated ExoPlayer 2.17.1 -> 2.18.0 ([changelog](https://github.com/google/ExoPlayer/blob/release-v2/RELEASENOTES.md))

I tested on my device and everything seemed to work fine. Feel free to test yourselves, though.

#### APK testing 
The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

Stypox commented:
You might have had some rebasing issues, since some of the changes are unrelated and undo some details of #8153

TacoTheDank commented:
> You might have had some rebasing issues, since some of the changes are unrelated and undo some details of #8153

I always keep my branches up-to-date; it simply looks like some PRs were merged while I was initially making this PR. No rebasing issues here! I have updated the PR.

All changes in this PR are to fix new ExoPlayer deprecations. I have double checked them to be sure.

I have also updated the YoutubeHttpDataSource note to 2.18.0: there were ([no real changes in upstream](https://github.com/google/ExoPlayer/commits/dev-v2/library/datasource/src/main/java/com/google/android/exoplayer2/upstream/DefaultHttpDataSource.java)). Just as a note, this class will need to be kept up-to-date with future upstream changes if (and inevitably when) they occur.

Redirion commented:
thank you for the PR! Please make sure that we opt out of the diagnostics with ExoPlayer.Builder.setUsePlatformDiagnostics(false)

around line [515](https://github.com/TeamNewPipe/NewPipe/blob/1404581e9b94ca85e50371dc700ffd22e204e19d/app/src/main/java/org/schabi/newpipe/player/Player.java#L515)

TacoTheDank commented:
Done 👍

Redirion reviewed (approved):
Tested Play, Pause, Mediabutton, Subtitles. Looks all good to me. Thanks again

mhmdanas commented on app/src/main/java/org/schabi/newpipe/player/datasource/YoutubeHttpDataSource.java:
```diff
@@ -1,5 +1,5 @@
 /*
- * Based on ExoPlayer's DefaultHttpDataSource, version 2.17.1.
+ * Based on ExoPlayer's DefaultHttpDataSource, version 2.18.0.
```
I don't think there's any point in this change?

AudricV replied:
See https://github.com/TeamNewPipe/NewPipe/pull/8536#issuecomment-1166390835. Anyway, this class will be removed soon in a future PR, which will use OkHttp as a network stack for playback and update the library on the fly.

mhmdanas replied:
Thanks for pointing that out, I didn't notice the comment.

TacoTheDank commented:
Will this be a potential issue?: https://github.com/google/ExoPlayer/issues/10381

Stypox commented:
It could, but we will probably have time to update to the next version before the next NewPipe release, so it shouldn't be a problem.

TacoTheDank commented:
Mm, hopefully.
